libs/model/src/contest/UpdateContestManagerMetadata.command.ts

```diff
@@ -0,0 +1,96 @@
+import type { Command } from '@hicommonwealth/core';
+import { InvalidState, schemas } from '@hicommonwealth/core';
+import { Op } from 'sequelize';
+import { z } from 'zod';
+import { models } from '../database';
+import { isCommunityAdmin } from '../middleware';
+import { mustExist } from '../middleware/guards';
+
+const Errors = {
+  InvalidTopics: 'Invalid topics',
+};
+
+export const UpdateContestManagerMetadata: Command<
+  typeof schemas.commands.UpdateContestManagerMetadata
+> = () => ({
+  ...schemas.commands.UpdateContestManagerMetadata,
+  auth: [isCommunityAdmin],
+  body: async ({ id, payload }) => {
+    const { topic_ids, ...rest } = payload;
+
+    const contestManager = await models.ContestManager.findOne({
+      where: {
+        community_id: id,
+        contest_address: payload.contest_address,
+      },
+    });
+
+    if (mustExist('ContestManager', contestManager)) {
+      let contestTopicsToCreate: z.infer<
+        typeof schemas.entities['ContestTopic']
+      >[] = [];
+
+      if (Array.isArray(topic_ids) && topic_ids.length > 0) {
+        const topics = await models.Topic.findAll({
+          where: {
+            id: {
+              [Op.in]: topic_ids,
+            },
+          },
+        });
+
+        if (topics.length !== topic_ids.length) {
+          throw new InvalidState(Errors.InvalidTopics);
+        }
+
+        contestTopicsToCreate = topics.map((t) => ({
```

Comment:
only adding new topics? guess we don't support removing? should we can this add_topics or something like that?

Reply:
not sure if we mean the same but in edit contest we can toggle topics on and off. Nut my understanding is that on every update all old topic associations are vanished and thats why we always create ContestTopic?

<img width="782" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14819225/c2bed089-0fa7-4497-9d12-d903c42fce55">